gRPC/src/main/kotlin/rys/ajaxpetproject/grpc/ChatFindAllService.kt

```diff
@@ -0,0 +1,46 @@
+package rys.ajaxpetproject.grpc
+
+import net.devh.boot.grpc.server.service.GrpcService
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
+import rys.ajaxpetproject.model.MongoChat
+import rys.ajaxpetproject.request.findOne.create.proto.ChatFindOneRequest
+import rys.ajaxpetproject.request.findOne.create.proto.ChatFindOneResponse
+import rys.ajaxpetproject.request.findOne.create.proto.ReactorChatFindAllServiceGrpc
+import rys.ajaxpetproject.service.ChatService
+import rys.ajaxpetproject.utils.toProto
+
+@GrpcService
+class ChatFindAllService(private val chatService: ChatService) :
+    ReactorChatFindAllServiceGrpc.ChatFindAllServiceImplBase() {
+
+    override fun chatFindAll(request: ChatFindOneRequest?): Flux<ChatFindOneResponse> {
+        return chatService.findAll()
+            .flatMap { item ->
+                buildSuccessResponse(item.toMono())
+                    .onErrorResume { e -> Mono.just(buildFailureResponse(e)) }
```

Comment:
what if chatService.findAll() produces error?

Reply:
Works fine! 
Tested. If error occurs then next emitted item will be FailureResponse and stream will be alive.

Reply:
Changed a bit